src/ts/ssh/secureStream.ts

```diff
@@ -182,12 +217,41 @@ export class SecureStream extends Duplex implements Disposable {
 			this.connectCompletion.resolve(this.stream);
 		} catch (e) {
 			if (!(e instanceof Error)) throw e;
-			let disconnectReason = e instanceof SshConnectionError ? e.reason : undefined;
-			disconnectReason ??= SshDisconnectReason.protocolError;
-			await this.session.close(disconnectReason, e.message, e);
-			this.connectCompletion.reject(e);
-			throw e;
+			if (e instanceof ObjectDisposedError && this.session.isClosed) {
+				// The session was closed while waiting for the channel.
+				// This can happen in reconnect scenarios.
+				this.connectCompletion.resolve(null);
+			} else {
+				let disconnectReason = e instanceof SshConnectionError ? e.reason : undefined;
+				disconnectReason ??= SshDisconnectReason.protocolError;
+				await this.session.close(disconnectReason, e.message, e);
+				this.connectCompletion.reject(e);
+				throw e;
+			}
+		}
+	}
+
+	/**
+	 * Re-initiates the SSH session over a NEW transport stream by exchanging initial messages
+	 * with the remote server. Waits for the secure reconnect handshake to complete. Additional
+	 * message processing is kicked off as a background task chain.
+	 *
+	 * Applies only to a secure stream client. (The secure stream server handles reconnections
+	 * automatically during the session handshake.)
+	 */
+	public async reconnect(transportStream: Stream | Duplex, cancellation?: CancellationToken) {
+		if (!(this.session instanceof SshClientSession)) {
+			throw new Error('Cannot reconnect SecureStream server.');
+		}
+
+		if (!transportStream) throw new TypeError('A transport stream is required.');
+		this.transportStream = transportStream;
```

Comment:
Should we check if the session is disconnected before trying to reconnect? Also should we close the previous `transportStream`?

Reply:
In a disconnected state, the previous stream would already be closed.

`this.session.reconnect()` will throw if not in disconnected state.